diff_drive_controller:
  type: "diff_drive_controller/DiffDriveController"
  publish_rate: 50

  left_wheel: ['front_left_wheel_joint', 'back_left_wheel_joint']
  right_wheel: ['front_right_wheel_joint', 'back_right_wheel_joint']

  left_wheel_radius: 0.121
  right_wheel_radius: 0.121
  wheel_seperation: 1

  # Odometry covariances for the encoder output of the robot. These values should
  # be tuned to your robot's sample odometry data, but these values are a good place
  # to start
  pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
  twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]

  # Top level frame (link) of the robot description
  base_frame_id: base_link
  enable_odom_tf: false
  # Velocity and acceleration limits for the robot
  linear:
    x:
      has_velocity_limits: false
      has_acceleration_limits: true
      max_acceleration: 0.6   # m/s^2
  angular:
    z:
      has_velocity_limits: false
      has_acceleration_limits: true
      max_acceleration: 6.0   # rad/s^2


front_left_wheel_joint_controller:
  type:  "effort_controllers/JointVelocityController"
  joint: "front_left_wheel_joint"
  pid: {p: 100.0, i: 0.0, d: 0.0}
back_left_wheel_joint_controller:
  type:  "effort_controllers/JointVelocityController"
  joint: "back_left_wheel_joint"
  pid: {p: 100.0, i: 0.0, d: 0.0}
front_right_wheel_joint_controller:
  type:  "effort_controllers/JointVelocityController"
  joint: "front_right_wheel_joint"
  pid: {p: 100.0, i: 0.0, d: 0.0}
back_right_wheel_joint_controller:
  type:  "effort_controllers/JointVelocityController"
  joint: "back_right_wheel_joint"
  pid: {p: 100.0, i: 0.0, d: 0.0}


joint_state_controller:
  # The joint state controller handles publishing transforms for any moving joints
  type: "joint_state_controller/JointStateController"
  publish_rate: 50

shoulder_joint_controller:
  type: "effort_controllers/JointEffortController"
  joint: shoulder
#  pid: {p: 100.0, i: 0.0, d: 0.0}

elbow_joint_controller:
  type: "effort_controllers/JointEffortController"
  joint: elbow
#  pid: {p: 100.0, i: 0.0, d: 0.0}

wrist_pitch_joint_controller:
  type: "effort_controllers/JointEffortController"
  joint: wrist_pitch
#  pid: {p: 100.0, i: 0.0, d: 0.0}

wrist_roll_joint_controller:
  type: "effort_controllers/JointEffortController"
  joint: wrist_roll
#  pid: {p: 100.0, i: 0.0, d: 0.0}
